iroh/src/node.rs

```diff
@@ -453,6 +450,7 @@ impl Callbacks {
         self.0.write().await.push(cb);
     }
 
+    #[allow(dead_code)]
```

Comment:
if you build it with no features, this is never used, and you get a clippy warning.

Reply:
The only place where send gets called is here, in provide0

```
        self.inner
            .callbacks
            .send(Event::ByteProvide(
                iroh_bytes::provider::Event::CollectionAdded { hash },
            ))
            .await;
```

but this part is disabled when the flat-db feature is off, because we don't have generic write yet. So it is unused when compiling wiht --no-default-features